cpp/dolfin/io/HDF5Interface.cpp

```diff
@@ -35,34 +34,44 @@ hid_t HDF5Interface::open_file(MPI_Comm mpi_comm, const std::string filename,
   }
 #endif
 
+  // Query file existence only once
+  bool file_exists = boost::filesystem::exists(filename);
```

Comment:
Is this needed? Is it in the right place? It's not relevant for wrote mode, and in other cases will the HDF5 error code pick up the issue?

Reply:
@garth-wells It is needed if we do not want to change HDF error handling in this PR. I tried to move responsibility for filesystem checks to HDF, but it prints error to stack, even if we just probe to open, otherwise create.

I can 
1.) Turn off error stack printing globally. Then handle error stack for every failure check.
2.) Temporarily turn stack off, then turn on after this probe. (But I am having difficulties finding default stack handler).

Reply:
We can have the check, but it's not needed in write mode (or append mode, if the fallback it to create a new file, which may or may not be a good thing).